### PR TITLE
chore(deps): bump most libp2p dependencies

### PR DESCRIPTION
**Pull request by qdm12**
## Changes

Unblocks #2768 

- `github.com/libp2p/go-libp2p-core/*` -> `github.com/libp2p/go-libp2p/core/*`
- `github.com/libp2p/go-libp2p-peerstore/pstoreds` -> `github.com/libp2p/go-libp2p/p2p/host/peerstore/pstoreds`
- `github.com/libp2p/go-libp2p-discovery` -> `github.com/libp2p/go-libp2p/p2p/discovery/routing`
- Fix btcsuite/btcd module issues
- Update mocks

## Tests

```sh
go test -tags integration github.com/ChainSafe/gossamer
```

## Issues

Unblocks #2768 

## Primary Reviewer

@jimjbrettj

### CODE REVIEW

File: dot/network/host.go [hunk 1]
```diff
@@ -159,7 +159,7 @@ func newHost(ctx context.Context, cfg *Config) (*host, error) {
 		return nil, err
 	}
 
-	ps, err := pstoreds.NewPeerstore(ctx, ds, pstoreds.DefaultOpts())
+	ps, err := pstoremem.NewPeerstore(ctx, ds, pstoremem.WithMaxProtocols(1024))
```
Comment (qdm12) [inline dot/network/host.go:162]: this is the only option left available from the default options of the deprecated implementation

**Review by EclesioMeloJunior — APPROVED** [review]
Nice stuff! 🎉  Those upgrades enable Gossamer to use go1.19?

**Review by jimjbrettj — APPROVED** [review]
Nice work!

**Comment by qdm12** [comment]
> Nice stuff! 🎉 Those upgrades enable Gossamer to use go1.19?

Yes, see #2768

**Pull request merged**